statusHandler/prometheusHandler.go

```diff
@@ -0,0 +1,105 @@
+package statusHandler
+
+import (
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/prometheus/client_golang/prometheus"
+	prometheusUtils "github.com/prometheus/client_golang/prometheus/testutil"
+)
+
+// PrometheusStatusHandler will define the handler which will update prometheus metrics
+type PrometheusStatusHandler struct {
+}
+
+var prometheusGaugeMetrics map[string]prometheus.Gauge
```

Comment:
fixed by using a sync map

Reply:
It should actually be a property on `PrometheusStatusHandler`, you still have race conditions because it's a global package variable. But you can solve them by moving `prometheusGaugeMetrics sync.Map` into the `PrometheusStatusHandler` struct

Reply:
ok, moved it